app/src/main/java/com/example/medapp/MainActivity.java

```diff
@@ -2,33 +2,131 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.appcompat.widget.Toolbar;
+import butterknife.BindView;
+import butterknife.ButterKnife;
 
+import android.content.DialogInterface;
 import android.os.Bundle;
 import android.view.Menu;
 import android.view.MenuInflater;
 import android.view.MenuItem;
+import android.view.View;
 import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.CheckBox;
 import android.widget.Spinner;
+import android.widget.Toast;
+import androidx.appcompat.app.AlertDialog;
 
 public class MainActivity extends AppCompatActivity {
-    Spinner sexSpinner;
+
+    @BindView(R.id.sex_spinner) Spinner sexSpinner;
+    @BindView(R.id.calculate)
+    Button calculateButton;
+    @BindView(R.id.sinus_checkbox)
+    CheckBox sinusCheckbox;
+    @BindView(R.id.arrhy_checkbox)
+    CheckBox arrhyCheckbox;
+    @BindView(R.id.conduction_checkbox)
+    CheckBox conductionCheckbox;
+    @BindView(R.id.left_checkbox)
+    CheckBox leftCheckbox;
+    int totalScore = 0;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        ButterKnife.bind(this);
 
         Toolbar toolbar = findViewById(R.id.tool_bar);
         toolbar.setTitle("MedApp");
         toolbar.setLogo(R.drawable.icons8_medical_doctor_48);
 
         setSupportActionBar(toolbar);
 
-                sexSpinner = findViewById(R.id.sex_spinner);
         String[] items = new String[]{"Male", "Female", "Binary"};
         ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_dropdown_item, items);
         sexSpinner.setAdapter(adapter);
 
+        calculateButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if(!sinusCheckbox.isChecked() && !arrhyCheckbox.isChecked() && !conductionCheckbox.isChecked() && !leftCheckbox.isChecked()) {
+                   //TODO Replace with a snackbar
+                    Toast.makeText(MainActivity.this, "All fields can't be empty", Toast.LENGTH_SHORT).show();
+                    return;
+                }
+                if(sinusCheckbox.isChecked()) {
+                    totalScore+=2;
+                }
+                if(arrhyCheckbox.isChecked()) {
+                    totalScore+=2;
+                }
+                if(conductionCheckbox.isChecked()) {
+                    totalScore+=1;
+                }
+                if(leftCheckbox.isChecked()) {
+                    totalScore+=1;
+                }
+
+                DisplayResult(totalScore);
+
+            }
+        });
+
+    }
+    //This display the heart failure percentage, summary and advise.
+    //An AlertBuilder is used for the display
+    private void DisplayResult(int result) {
+        String summaryMessage = "";
```

Comment:
clean up this logic, you can create a nested object or map. presently it is not visible. You can move it to a viewModel or presenter.

Reply:
Can you please explain further. I don't understand very well.

Reply:
business logic should not be in the view, move it to a presenter. check MVP or use viewModel MVVM

Reply:
Okay, I will check that out